### PR TITLE
Signal processing logging

### PR DESCRIPTION
**Pull request by treitter**

### CODE REVIEW

**Review by araujol — COMMENTED** [review]
Regarding reformatting the tests, I agree this indeed makes the expected output much more readable, though I also would suggest to assign these values to global variables or group them into a list/dictionary that they can be passed as variables to the tests functions, and that would make the functions indentation easier to read too.

File: vsm [hunk 1]
```diff
@@ -19,6 +19,8 @@ import threading
 import time
 from ipc.loader import load_plugin, LoaderError
 import os
+# FIXME: cut
+import inspect
```
Comment (araujol) [inline vsm]: Why is **inspect** imported here?

Comment (treitter) [reply vsm]: That was supposed to be removed. I've fixed it in the current version of the branch (just pushed).

**Comment by treitter** [comment]
Do you mean creating variables like this?:

```
state_moving = '''
State = {
moving = True
}
'''
```

If that's the case, I think there wouldn't be much room for re-use. If we moved them globally, it would clean up the indentation slightly but I'm not sure it would be worth having to separate the expected result strings from their taste cases.

**Comment by araujol** [comment]
Yes, that's what I meant. It was only for indentation purpose, but besides that, no other benefit from it, so I am fine with keeping the strings in the test cases too.

The code looks good to me, and If the tests pass fine for the recently merged IPC tests, then it is good to merge.

**Comment by treitter** [comment]
Yes, the tests pass, so I've merged this.

**Pull request merged**